src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2190,7 +2126,7 @@ public override BoundNode VisitNullCoalescingAssignmentOperator(BoundNullCoalesc
             var leftState = this.State.Clone();
             LearnFromNonNullTest(leftOperand, ref leftState);
             LearnFromNullTest(leftOperand, ref this.State);
-            TypeWithState rightResult = VisitOptionalImplicitConversion(rightOperand, targetType, UseLegacyWarnings(leftOperand), AssignmentKind.Assignment);
+            TypeWithState rightResult = VisitOptionalImplicitConversion(rightOperand, targetType, useLegacyWarnings: UseLegacyWarnings(leftOperand), trackNullability: false, AssignmentKind.Assignment);
```

Comment:
>trackNullability: false, [](start = 149, length = 24)

I'm unsure whether this should be be `trackNullability: true` (I assume we'd remove the call to `TrackNullableStateForAssignment` on the next line)